sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncDatabase.java

```diff
@@ -663,6 +686,78 @@ public CosmosAsyncUser getUser(String id) {
                                                                                               .getThroughput()));
     }
 
+    /**
+     * Sets throughput provisioned for a container in measurement of
+     * Requests-per-Unit in the Azure Cosmos service.
+     *
+     * @param throughputProperties the throughput properties
+     * @return the mono
+     */
+    public Mono<ThroughputResponse> replaceThroughput(ThroughputProperties throughputProperties) {
+        return this.read()
+                   .flatMap(response -> this.getDocClientWrapper()
+                                            .queryOffers(getOfferQuerySpecFromResourceId(response.getProperties()
+                                                                                             .getResourceId()),
+                                                         new FeedOptions())
+                                            .single()
+                                            .flatMap(offerFeedResponse -> {
+                                                if (offerFeedResponse.getResults().isEmpty()) {
+                                                    return Mono.error(BridgeInternal
+                                                                          .createCosmosClientException(
+                                                                              HttpConstants.StatusCodes.BADREQUEST,
+                                                                              "No offers found for the " +
+                                                                                  "resource"));
```

Comment:
it is worth including the name of the database in the sdk generated error message.

Reply:
@j82w lets please track it part of diagnostics improvement.

Reply:
Added resource name